src/Workspaces/Core/Portable/Rename/ConflictEngine/RelatedLocation.cs

```diff
@@ -2,31 +2,71 @@
-        public RelatedLocationType Type { get; set; }
-        public bool IsReference { get; }
-        public DocumentId DocumentId { get; }
+        /// <summary>
+        /// The Span of the original identifier if it was in source, otherwise the span to check for implicit
+        /// references.
+        /// </summary>
+        public readonly TextSpan ConflictCheckSpan;
+        public readonly RelatedLocationType Type;
+        public readonly bool IsReference;
+        public readonly DocumentId DocumentId;
 
-        // If there was a conflict at ConflictCheckSpan during rename, then the next phase in rename uses ComplexifiedTargetSpan span to be expanded to resolve the conflict
-        public TextSpan ComplexifiedTargetSpan { get; }
+        /// <summary>
+        /// If there was a conflict at ConflictCheckSpan during rename, then the next phase in rename uses
+        /// ComplexifiedTargetSpan span to be expanded to resolve the conflict.
+        /// </summary>
+        public readonly TextSpan ComplexifiedTargetSpan;
 
-        public RelatedLocation(TextSpan location, DocumentId documentId, RelatedLocationType type, bool isReference = false, TextSpan complexifiedTargetSpan = default)
+        public RelatedLocation(TextSpan conflictCheckSpan, DocumentId documentId, RelatedLocationType type, bool isReference = false, TextSpan complexifiedTargetSpan = default)
         {
-            this.ConflictCheckSpan = location;
+            this.ConflictCheckSpan = conflictCheckSpan;
             this.Type = type;
             this.IsReference = isReference;
             this.DocumentId = documentId;
             this.ComplexifiedTargetSpan = complexifiedTargetSpan;
         }
+
+        public RelatedLocation WithType(RelatedLocationType type)
+            => new RelatedLocation(ConflictCheckSpan, DocumentId, type, IsReference, ComplexifiedTargetSpan);
+
+        public override bool Equals(object obj)
+            => obj is RelatedLocation location && Equals(location);
+
+        public bool Equals(RelatedLocation other)
+        {
+            return ConflictCheckSpan.Equals(other.ConflictCheckSpan) &&
+                   Type == other.Type &&
+                   IsReference == other.IsReference &&
+                   EqualityComparer<DocumentId>.Default.Equals(DocumentId, other.DocumentId) &&
+                   ComplexifiedTargetSpan.Equals(other.ComplexifiedTargetSpan);
+        }
+
+        public override int GetHashCode()
+        {
+            var hashCode = 928418920;
+            hashCode = hashCode * -1521134295 + ConflictCheckSpan.GetHashCode();
+            hashCode = hashCode * -1521134295 + Type.GetHashCode();
+            hashCode = hashCode * -1521134295 + IsReference.GetHashCode();
+            hashCode = hashCode * -1521134295 + EqualityComparer<DocumentId>.Default.GetHashCode(DocumentId);
+            hashCode = hashCode * -1521134295 + ComplexifiedTargetSpan.GetHashCode();
+            return hashCode;
```

Comment:
Use Hash.Combine?

Reply:
i can't.  not vailable on netstandard.

Reply:
http://sourceroslyn.io/#Microsoft.CodeAnalysis/InternalUtilities/Hash.cs,14

Reply:
sure.  i can do that in a followup.

Note: the benefit of the above approach is that the IDE detects it and will offer to upgrade it once you do have access to System.HashCode.